### DISCUSSION TITLE
Change Row Click Action to Open View Modal

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#4707] [Help] [answered]
Is it possible to change the click action for a table row to open the View Modal, rather than navigating to the edit page?

**Comment by Z3d0X** [comment, marked as answer]
On your Resource:

- Make sure ViewAction exists on your table

On LW Component (ListRecord Page):
```php
protected function getTableRecordUrlUsing(): ?Closure
{
    return null;
}

protected function getTableRecordActionUsing(): ?Closure
{
    return fn () => 'view';
}
 ```

**Reply by intrepidws** [reply]
Thanks for your help.

**Comment by mattlibera** [comment]
for v3 i was banging my head on this for a while, and the above did not work. but a colleague of mine found that this did, in the table declaration itself (on the resource file):

```
return $table
    // .... your schema...
    ->recordAction(Tables\Actions\ViewAction::class)
    ->recordUrl(null);
```

**Comment by fidelix** [comment]
My god I searched for this so much, it really should be in the docs.
